lib/jekyll-admin/server.rb

```diff
@@ -59,7 +59,8 @@ def front_matter
 
     def document_body
       body = if front_matter && !front_matter.empty?
-               YAML.dump(front_matter).strip
+               YAML.dump(restored_front_matter).strip
+                 .gsub(": 'null'", ": null") # restore null values
```

Comment:
Yes `JSON.parse` at front end (via your patch) catches it fine but ends up as `nil` when `JSON.parse` runs again on the `request_body` here which gets `dumped` by `YAML` as an empty value.

i.e.
JSON `{"layout": null, "foo": false}` ==> Ruby Hash `{"layout" => "nil", "foo" => false}` ==> Front Matter:
```
---
layout:
foo: false
---
```